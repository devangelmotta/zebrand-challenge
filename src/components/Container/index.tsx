import React, { memo } from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: ${(props: Props) => props.display || "flex"};
  position: ${props => props.position || "relative"};
  flex-direction: ${props => props.flexDirection || "column"};
  justify-content: ${props => props.justifyContent || "space-between"};
  align-content: ${props => props.alignContent || "center"};
  align-items: ${props => props.alignItems || "center"};
  width: ${props => props.width || "100%"};
  max-width: ${props => props.maxWidth || "none"};
  min-width: ${props => props.minWidth || "none"};
  height: ${props => props.height || "auto"};
  min-height: ${props => props.maxHeight || "none"};
  max-height: ${props => props.minHeight || "none"};
  background-color: ${props => props.backgroundColor || "transparent"};
  background: ${props => props.backgroundColor || ""};
  padding: ${props => props.padding || "0 0 0 0"};
  border: ${props => props.border || "0 solid transparent"};
  border-radius: ${props => props.borderRadius || "0"};
  margin: ${props => props.margin || "0 0 0 0"};
  z-index: ${props => props.zIndex || "auto"};
  -webkit-box-shadow: ${props => props.shadow || "0px 0px 0px 0px rgba(0,0,0,0)"}; 
  -moz-box-shadow: ${props => props.shadow || "0px 0px 0px 0px rgba(0,0,0,0)"}; 
  box-shadow: ${props => props.shadow || "0px 0px 0px 0px rgba(0,0,0,0)"}; 
  `;

interface Props {
  readonly display? : string,
  readonly flexDirection? : string,
  readonly justifyContent? : string,
  readonly position? : string,
  readonly alignContent? : string,
  readonly alignItems? : string,
  readonly width? : string,
  readonly height? : string,
  readonly minHeight? : string,
  readonly maxHeight? : string,
  readonly minWidth? : string,
  readonly maxWidth? : string,
  readonly backgroundColor? : string,
  readonly background? : string,
  readonly padding? : string,
  readonly border? : string,
  readonly borderRadius? : string,
  readonly margin? : string,
  readonly hoverBgColor? : string,
  readonly hoverBorder? : string,
  readonly zIndex?: string
  readonly shadow?: string;
}

export default memo(Container);

