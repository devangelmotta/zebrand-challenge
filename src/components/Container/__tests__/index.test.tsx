import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import Container from '..';

test('it works', () => {
  const tree = renderer.create(<Container />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('display', 'flex')
  expect(tree).toHaveStyleRule('position', 'relative')
  expect(tree).toHaveStyleRule('flex-direction', 'column')
  expect(tree).toHaveStyleRule('justify-content', 'space-between')
  expect(tree).toHaveStyleRule('align-content', 'center')
  expect(tree).toHaveStyleRule('align-items', 'center')
  expect(tree).toHaveStyleRule('width', '100%')
  expect(tree).toHaveStyleRule('max-width', 'none')
  expect(tree).toHaveStyleRule('min-width', 'none')
  expect(tree).toHaveStyleRule('height', 'auto')
  expect(tree).toHaveStyleRule('max-height', 'none')
  expect(tree).toHaveStyleRule('background-color', 'transparent')
  expect(tree).toHaveStyleRule('padding', '0 0 0 0')
  expect(tree).toHaveStyleRule('border', '0 solid transparent')
  expect(tree).toHaveStyleRule('border-radius', '0')
  expect(tree).toHaveStyleRule('margin', '0 0 0 0')
  expect(tree).toHaveStyleRule('z-index', 'auto')
  expect(tree).toHaveStyleRule('-webkit-box-shadow', '0px 0px 0px 0px rgba(0,0,0,0)')
  expect(tree).toHaveStyleRule('-moz-box-shadow', '0px 0px 0px 0px rgba(0,0,0,0)')
  expect(tree).toHaveStyleRule('box-shadow', '0px 0px 0px 0px rgba(0,0,0,0)')
})
