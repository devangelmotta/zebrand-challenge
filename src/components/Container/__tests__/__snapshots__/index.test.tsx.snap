// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: none;
  min-width: none;
  height: auto;
  max-height: none;
  background-color: transparent;
  padding: 0 0 0 0;
  border: 0 solid transparent;
  border-radius: 0;
  margin: 0 0 0 0;
  z-index: auto;
  -webkit-box-shadow: 0px 0px 0px 0px rgba(0,0,0,0);
  -moz-box-shadow: 0px 0px 0px 0px rgba(0,0,0,0);
  box-shadow: 0px 0px 0px 0px rgba(0,0,0,0);
}

<div
  className="c0"
/>
`;
